### PR TITLE
Elixir 1.12 otp 24 preparation

### PR DESCRIPTION
**Pull request by slashmili**
* :crypto.hmac is  deprecated in OTP 24
* :crypto.block_encrypt/3 is  deprecated in OTP 24
* `ECPrivateKey` record has a new item in OTP 24
* Updated test dependencies  
* Removed Timex 🎉

### CODE REVIEW

**Comment by slashmili** [comment]
Well the `:crypto.mac/4` is not available in OTP 22 which we do use 😢 

I might go with condition to check the Erlang version and run different function!

**Review by nonninz — APPROVED** [review]
LGTM 👍🏻 

Where do we use OTP22?

**Comment by slashmili** [comment]
> Where do we use OTP22?

in DC

**Comment by slashmili** [comment]
Just found out that `:crypto.block_encrypt/3` is also removed 😢 
> warning: :crypto.block_encrypt/3 is undefined or private, use crypto:crypto_one_time/4 or crypto:crypto_init/3 + crypto:crypto_update/2 + crypto:crypto_final/1 instead

File: lib/hm_crypto/access_certificate.ex [hunk 1]
```diff
@@ -175,11 +175,6 @@ defmodule HmCrypto.AccessCertificate do
   end
 
   defp encode_date(date) do
-    date
-    |> Timex.Timezone.convert("Etc/UTC")
-    |> Timex.format!("%y,%m,%d,%H,%M", :strftime)
-    |> String.split(",")
-    |> Enum.map(&String.to_integer/1)
-    |> :binary.list_to_bin()
+    :binary.list_to_bin([date.year - 2000, date.month, date.day, date.hour, date.minute])
```
Comment (nonninz) [inline lib/hm_crypto/access_certificate.ex:178]: Maybe no need for the `list_to_bin` extra step:

```suggestion
    <<date.year - 2000, date.month, date.day, date.hour, date.minute>>
```

**Review by nonninz — APPROVED** [review]
LGTM! 👍🏻 With minor change suggestion

**Comment by slashmili** [comment]
We are good to upgrade our Erlang nodes to OTP 24 🎉

**Pull request merged**